ggshield/cmd/secret/scan/secret_scan_common_options.py

```diff
@@ -126,6 +129,31 @@ def _banlist_detectors_callback(
     metavar="DETECTOR",
 )
 
+
+def _with_incident_details_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[bool]
+) -> Optional[bool]:
+    if value is not None:
+        ctx_obj = ContextObj.get(ctx)
+        ctx_obj.client = create_client_from_config(ctx_obj.config)
+        response = ctx_obj.client.api_tokens()
+
+        if not isinstance(response, (Detail, ApiTokensResponse)):
+            raise UnexpectedError("Unexpected api_tokens response")
+        elif isinstance(response, Detail) and response.status_code == 401:
+            raise APIKeyCheckError(ctx_obj.client.base_uri, "Invalid API key.")
+        elif isinstance(response, Detail):
+            raise UnexpectedError(response.detail)
+
+        if TokenScopes.INCIDENTS_READ not in response.scopes:
+            raise MissingScopeError(
+                "Token is missing the scope: 'incidents:read' to retrieve incident details."
+            )
+
+        ctx_obj.config.user_config.secret.with_incident_details = value
+    return value
```

Comment:
It's not a good idea to create a client and send HTTP requests while parsing the command-line because some not-yet-parsed CLI flags might affect it, for example  `--allow-self-signed`.

In general we should try to run as little code as possible while parsing the command-line.

You should be able to move the logic of your check to `SecretScanner.__init__()`.